src/main/java/uk/gov/register/core/TransactionalMemoizationStoreImpl.java

```diff
@@ -0,0 +1,68 @@
+package uk.gov.register.core;
+
+import uk.gov.verifiablelog.store.memoization.MemoizationStore;
+import java.util.HashMap;
+
+public class TransactionalMemoizationStoreImpl implements TransactionalMemoizationStore {
+    private final MemoizationStore instance;
+    private final HashMap<Integer, HashMap<Integer, byte[]>> temporaryEntries;
+    private Integer currentEntryCount;
+
+    public TransactionalMemoizationStoreImpl(
+            MemoizationStore instance) {
```

Comment:
can we rename `instance` to something else pls it's not very descriptive, suggests it's a self instance and not the decorated one.